### DISCUSSION TITLE
Confused on branching structure

### DISCUSSION THREAD

**Discussion by 0xArchiviste** [ethereum/web3.js#4887] [Q&A] [answered]
I noticed releases are done on the 1.x branch but I am noticing a lot of really important changes happening on the 4.x branch. How exactly are these incorporated into releases if at all? Do things from 4.x eventually merge back into 1.x.

In the event that is not true:

- Can we actively pull and use 4.x?
- Is there a 4.x release planned any time soon?

Sorry if this is a terrible question

**Comment by luu-alex** [comment, marked as answer]
Hey there @Sal-Ali, Thank you for creating this post as this is not a terrible question, feel free to read this [article](https://blog.chainsafe.io/web3-js-solid-foundations-create-stable-structures-1674514fe35)  :)
1.x is an older library and maintaining it sometimes proves to be difficult making changes as it relies on some older packages. We still release with changes that are neccesary in 1.x as we still maintain the library and merge community PRs but we also are working hard on a 4.x typescript rewrite so that we can have a better codebase while still having parity with 1.x. 
 - You cannot actively pull 4.x yet, we'll have an alpha version for 4.x soon 
 - You can see the release schedule [here](https://github.com/ChainSafe/web3.js/blob/1.x/RELEASE.md#long-term-support-lts)

**Reply by 0xArchiviste** [reply]
Thank you for the insight, much appreciated. Excited for 4.x!
